### PR TITLE
feat(utilities): add retry function

### PR DESCRIPTION
**Pull request by favna**
Thanks goes to @kyranet

### CODE REVIEW

**Review by vladfrangu — DISMISSED** [review]

File: packages/utilities/src/lib/retry.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import type { Awaitable } from './types';
+
+/**
+ * This code is used to retry a function call a specified number of times.
+ * The function will retry the call to the callback function until it either succeeds or it runs out of retries.
+ * @param  cb The function to be retried is passed in as a callback function.
+ * @param  retries The number of retries is also passed in as a parameter.
```
Comment (kyranet) [inline packages/utilities/src/lib/retry.ts:7]:
```suggestion
 * @param cb The function to be retried is passed in as a callback function.
 * @param retries The number of retries is also passed in as a parameter.
```

File: packages/utilities/src/lib/retry.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import type { Awaitable } from './types';
+
+/**
+ * This code is used to retry a function call a specified number of times.
+ * The function will retry the call to the callback function until it either succeeds or it runs out of retries.
```
Comment (kyranet) [inline packages/utilities/src/lib/retry.ts:5]:
```suggestion
 * Calls the callback function until it either succeeds or it runs out of retries.
```

File: packages/utilities/tests/retry.test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { retry } from '../src/lib/retry';
+
+describe('retry', () => {
+	test('GIVEN a simple string return THEN returns the same on first attempt', async () => {
+		const result = await retry(() => 'test', 3);
```
Comment (kyranet) [inline packages/utilities/tests/retry.test.ts:5]: If we use `vi.fn`, we can test how many times the function is called to ensure the amount is correct.

File: packages/utilities/tests/retry.test.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+		expect(result).toBe('test');
+	});
+
+	test('GIVEN a twice failing function THEN returns the third result', async () => {
+		let counter = 0;
+		const cb = () => {
```
Comment (kyranet) [inline packages/utilities/tests/retry.test.ts:12]: Ditto

**Review by kyranet — CHANGES_REQUESTED** [review]
I wonder if we should have a `retrySync` (or a sync `retry` with `retryAsync`). 🤔

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — DISMISSED** [review]

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — CHANGES_REQUESTED** [review]
Forgot to re-export in index.ts :megu:

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**